tremor-script/src/interpreter.rs

```diff
@@ -693,99 +741,138 @@ where
     'event: 'run,
 {
     let patch_expr = expr;
+    let mut evaluated: Vec<PreEvaluatedPatchOperation> = Vec::with_capacity(expr.operations.len());
+    // first pass over the operations, evaluating them
+    // and (IMPORTANT) get it into an owned, possibly cloned value, so we reference
+    // the target value in the state before any patch operation has been executed.
     for op in &expr.operations {
-        // NOTE: This if is inside the for loop to prevent obj to be updated
-        // between iterations and possibly lead to dangling pointers
+        evaluated.push(match op {
```

Comment:
Perhaps you can extract the conversion from `PatchOperation` into `PreEvaluatedPatchOperation` as a method on either type. That might also fix of the clippy warning suppressed on line 725. The main blocker would be the need to pass in `opts`, `env`, `event`, `state`, `meta`, and `local` to another function. On the other hand, the Tremor code base doesn't shy away from using explicit `inline` annotations, so it might clean up the code base without runtime overhead.

Reply:
✅